__new_qenv.R

```diff
@@ -0,0 +1,316 @@
+
+#' qenv refactor prototype
+#'
+#' Simple to use environment with history tracking.
+#'
+#' @param data,x (`qenv`)
+#' @param expr (`language`) simple or compound expression to evaluate in `data`
+#' @param text (`character`) character vector of expressions to evaluate in `data`
+#' @param ... `name:value` pairs to inject values into `expr`
+#'
+#' @return
+#' `qenv` returns a `qenv` object. `with` returns NULL invisibly. `within` returns a modified deep copy of `data`.
+#'
+#' @describeIn qenv create `qenv` object
+#' @export
+qenv <- function(file) {
+  ans <- new.env()
+  attr(ans, "code") <- list()
+  attr(ans, "errors") <- list()
+  attr(ans, "warnings") <- list()
+  attr(ans, "messages") <- list()
+  class(ans) <- c("qenv", class(ans))
+  ans
+}
+
+
+#' @describeIn qenv act in `qenv` object
+#' @export
+with.qenv <- function(data, expr, text, ...) {
+  code <- .prepare_code(if (!missing(expr)) substitute(expr), if (!missing(text)) text)
```

Comment:
Would using `str2expression()` help simplify the processing of `text` expressions? There seem to be some issues with multi-line expression or compound expressions

Check these 2 examples (mocking reading the expression from file)

#### 1

```R
testthat::test_that("with.qenv Supports multiline text expression", {

  str_expr <- "1 +
  1"

  script_path <- withr::local_file(tempfile())
  writeLines(str_expr, script_path)

  q <- qenv()

  with(q, text = readLines(script_path)) |>
    testthat::expect_no_error()
})
```
#### 2

```R
testthat::test_that("with.qenv Supports multi-line expression", {

  str_expr <- "{
  1 +
  1
  2 * 3
}"

  script_path <- withr::local_file(tempfile())
  writeLines(str_expr, script_path)

  q <- qenv()

  with(q, text = readLines(script_path)) |>
    testthat::expect_no_error()
})
``